include/onnxruntime/core/framework/op_kernel_info.h

```diff
@@ -45,6 +44,10 @@ class OpKernelInfo : public OpNodeProtoHelper<ProtoHelperNodeContext> {
 
   bool TryGetConstantInput(int input_index, const Tensor** constant_input_value) const;
 
+  AllocatorPtr GetDeviceAllocator() const { return device_allocator_; }
+
+  AllocatorPtr GetPinnedAllocator() const { return pinned_allocator_; }
```

Comment:
I would prefer to keep the GetAllocator method. we don't want to limit the kernel only can get device / pinned allocator.